### ISSUE TITLE
[TODO]: Implement HPO and change default HPO search space

### ISSUE THREAD

**Issue by kevingreenman** [chemprop/chemprop#496]
Based on internal discussion about the performance of HPO on numerous projects, we intend to make the following changes once hyperparameter optimization is implemented in v2:
* `ffn_num_layers`:  `hyperopt.quniform` [1,3] --> [2,6]
* `dropout`: `hyperopt.quniform` [0,0.4] (increments of 0.05) --> [0,0.4] (draws dropout=0.0 more often in random search)
  * nest the dropout value choice inside a yes/no choice for whether there is any dropout at all

These are primarily intended to help give better performance in the initial random trials that seed the TPE algorithm in later trials.

**Comment by davidegraff** [comment]
I don't think we need to go crazy with parameterization here. It's easy for users to ask "can I input my own hyperparameter search spaces," but that would add a decent amount of complexity to this script. We also don't need to make the search space super-duper large. Rather, I think we should keep it focused on a few hyperparams: FFN layers, activation, learning rate, and message size. As long as we write the code for this well enough, we can always respond to users asking the above: "if you feel like you'd like to extend the builtin HPO capabilities, please see <THIS_FILE> for an example of how to do that and modify it accordingly." Or even better, make HPO a function in the CLI that users can easily import and use their own settings so that they can easily use custom search spaces without needing to reinvent our wheel.

**Comment by kevingreenman** [comment]
Note for whoever ends up implementing this in v2: #571 contains an example of the intended change, but written in the v1 code
